### PR TITLE
feat(edge): Ignore functions without typical loggroup

### PR DESCRIPTION
**Pull request by iDVB**
- Add filter to ensure log groups exist before subscribing
- Add ESLint
- Add Prettier
- Let Prettier refactor code layout
- Remove Babel build step for functions

### CODE REVIEW

**Comment by iDVB** [comment]
@simlu @MacMcIrish 
Sorry for the large commit/pr

I changed a few things as more of a suggestion to make things a lot cleaner. 

1) **ESLint + Prettier**
Since the promises we're very terse it made it very hard to read. Implementing these allows for a single format that all contributors can adhere to.

2) **Remove Babel build**
Since you are using Node 8.10 ... there was nothing in the codebase that even needed transpiling.
Additionally, transpiling results in code that can not easily be edited/tested from within the AWS console.  So it was basically just an extra step that was not needed and made things less portable.

3) **Check if Log Group exists**
[appendLogRetentionInfo](https://github.com/iDVB/lambda-monitor/blob/2df670bacb2b6f4e05e07547f5c895cf343f22e3/src/logic/util/lambda.js#L37-L38)
[appendLogSubscriptionInfo](https://github.com/iDVB/lambda-monitor/blob/2df670bacb2b6f4e05e07547f5c895cf343f22e3/src/logic/util/lambda.js#L54)

**Comment by simlu** [comment]
@iDVB We should have definitely chatted before! 

(1) This code base uses [js-gardener](https://github.com/blackflux/js-gardener), so it already enforces eslint (and a lot of other stuff)
(2) Adding prettier should probably happen in js-gardener (opposed to here).
(3) Babel is used so we can translate flow -> flow comment style (this allows us to use packages as plain js packages, but also as flow annotated packages)

Let's chat! There is a slack workspace.
https://join.slack.com/t/blackflux/shared_invite/enQtNDY2ODAxMTU1NTIzLWVkMzdhM2NhMGVjNzQ5YjRjZDk2OTBhODhlMmZjNGI4YWE0NzY5MDI3YzA2YTI1YmZhM2U1MDgxZDRkMzc0MTQ

**Comment by simlu** [comment]
@iDVB Looking through this, I think these should be two prs. First one to change formatting and second one to change functionality. This one is very hard to review.

PS: To ensure ci passes you can always run `npm t`.

File: src/logic/process-logs.js [hunk 1]
```diff
@@ -1,71 +1,119 @@
...
+            billedDuration: parseInt(requestLog[3], 10),
+            maxMemory: parseInt(requestLog[4], 10),
+            memory: parseInt(requestLog[5], 10),
+            env: process.env.STAGE,
+          });
+        } else if (
+          !logEvent.message.match(requestEndRegex) &&
+          !logEvent.message.match(requestStartRegex)
+        ) {
+          console.log('HEY', logEvent);
```
Comment (MacMcIrish) [inline src/logic/process-logs.js]: Was this meant to be committed? :slightly_smiling_face:

Comment (iDVB) [reply src/logic/process-logs.js]: No, this PR was more to just make things visible to you guys. 
I did not expect it to be ready to merge at this time.

**Pull request closed without merging**